pkg/clone/clone.go

```diff
@@ -0,0 +1,127 @@
+package clone
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"strings"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/storage/memory"
+
+	//nolint:staticcheck
+	"io/ioutil"
+	"net/http"
+	"net/url"
+)
+
+const baseURL = "http://kv.bacalhau.org/"
+
+type ScriptStruct struct {
+	//nolint:unused
+	path string
+	//nolint:unused
+	arguments []string
+}
+
+type Clone struct {
+	URL string
+}
+
+type Response struct {
+	CID string
+}
+
+func NewCloneClient() (*Clone, error) {
+	return &Clone{
+		URL: "",
+	}, nil
+}
+
+func RepoExistsOnIPFSGivenURL(urlStr string) (string, error) {
+	output, err := GetLatestCommitHash(urlStr)
+	if err != nil {
+		fmt.Println(err)
+		return "", err
+	}
+	url := baseURL + output
+	ctx := context.Background()
```

Comment:
This func needs to take a context.Context rather than using Background so that it can do proper logging and be cancelled.